src/ext/java/org/opentripplanner/ext/siri/SiriAlertsUpdateHandler.java

```diff
@@ -104,11 +104,15 @@ public void update(ServiceDelivery delivery) {
                         addedCounter++;
                         if (alert != null) {
                             alert.setId(situationNumber);
-                            alerts.removeIf(transitAlert -> transitAlert.getId().equals(situationNumber));
-                            alerts.add(alert);
                             if (alert.getEntities().isEmpty()) {
-                                LOG.info("No match found for Alert - ignoring situation with situationNumber {}", situationNumber);
+                                LOG.info(
+                                        "No match found for Alert - setting Unknown entity for situation with situationNumber {}",
+                                        situationNumber
+                                );
                             }
+                            alert.addEntity(new EntitySelector.Unknown("Alert had no entities that could be handled"));
```

Comment:
Maybe it would also be possible to add Unknown selector each time a selector was ignored because we don't handle it yet instead of adding one Unknown selector at the end like this.

Reply:
Should the "add unknown entity" be inside the if-block?

Reply:
Indeed... I'll fix that and update tests a bit so they will catch if suddenly new entities appear.

Reply:
Fixed it now and changed tests so that they test that only the intended entities get added instead of checking that x type entities don't get added.